lib/devise/models/confirmable.rb

```diff
@@ -169,6 +169,7 @@ def skip_reconfirmation!
         # in models to map to a nice sign up e-mail.
         def send_on_create_confirmation_instructions
           send_confirmation_instructions
+          skip_reconfirmation!
```

Comment:
@lmduc i've added #4156. I don't remove or change any tests there, so your code should work fine.